src/mjolnir/landmark_builder.cc

```diff
@@ -183,6 +194,38 @@ void LandmarkDatabase::insert_landmark(const std::string& name,
   pimpl->vacuum_analyze = true;
 }
 
+Landmark LandmarkDatabase::get_landmark(const uint32_t pkey) {
+  auto* get_landmark_stmt = pimpl->get_landmark_stmt;
+
+  sqlite3_reset(get_landmark_stmt);
+  sqlite3_clear_bindings(get_landmark_stmt);
+
+  sqlite3_bind_int(get_landmark_stmt, 1, pkey);
+  LOG_TRACE(sqlite3_expanded_sql(get_landmark_stmt));
+
+  int ret = sqlite3_step(get_landmark_stmt);
+  if (ret == SQLITE_ROW) {
+    uint32_t landmark_id = static_cast<uint32_t>(sqlite3_column_int(get_landmark_stmt, 0));
+    const char* name = reinterpret_cast<const char*>(sqlite3_column_text(get_landmark_stmt, 1));
+
+    int landmark_type = -1;
+    if (sqlite3_column_type(get_landmark_stmt, 2) != SQLITE_NULL) {
+      landmark_type = sqlite3_column_int(get_landmark_stmt, 2);
+    }
```

Comment:
this can't happen can it? if so, we should throw I think. passing on a value of `-1` would do crazy things.

Reply:
i agree, i think we can just skip the if completely and set the type without checking null. we dont ever put a null entry in the db so we cant ever get a null entry out

Reply:
got it